### PR TITLE
fix: instances of error within array should be nested in object property

### PR DESCRIPTION
**Pull request by reggi**

### CODE REVIEW

File: lib/deep-map.js [hunk 1]
```diff
@@ -8,20 +8,32 @@ function filterError (input) {
   }
 }
 
-const deepMap = (input, handler = v => v, path = ['$'], seen = new Set([input])) => {
+const deepMap = (
+  input,
+  handler = v => v,
+  path = ['$'],
+  seen = new Set([input]),
```
Comment (hashtagchris) [inline lib/deep-map.js:15]: 🤔 If `input` is an array, should we avoid doubly nesting it?

Comment (hashtagchris) [reply lib/deep-map.js:15]: Related, would it be easier to reason about this logic if we had the exported function do something like:

```
module.exports = function (input, etc) {
  if (Array.isArray(input)) {
    return input.map(e => deepMap(e, ...etc))
  }
  return deepMap(input, ...etc)
}
```

Comment (reggi) [reply lib/deep-map.js:15]: `Set` takes an array and an item in that array is an array, and we do a match on the instance of that array (input), to check if it is `seen`, I think `seen` is correct the way it is.

**Comment by hashtagchris** [comment]
As discussed offline, I think this logic is getting complicated, and it's fine to make the caller call `redact` separately for each logging argument. I'm okay with errors in an array not being nested.

**Pull request closed without merging**